src/magnify/apps/core/api.py

```diff
@@ -234,3 +202,125 @@ def perform_create(self, serializer):
         RoomUser.objects.create(
             room=room, user=self.request.user, is_administrator=True
         )
+
+    @action(
+        methods=["post"],
+        detail=True,
+        url_path="users",
+        permission_classes=[permissions.IsAuthenticated],
+    )
+    # pylint: disable=unused-argument,invalid-name
+    def users(self, request, pk=None):
+        """Adds a user in a room."""
+        room = self.get_object()
+        user = User.objects.get(id=request.data.get("user"))
+        serializer = RoomUserSerializer(
+            data={
+                "room": room.id,
+                "user": user.id,
+                "is_administrator": request.data.get("is_administrator"),
+            },
+            context={"request": request},
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=201, headers=headers)
+
+    @action(
+        methods=["get", "put", "delete"],
+        detail=True,
+        url_path="users/(?P<user_pk>[^/.]+)",
+        permission_classes=[IsRoomAdministrator],
+    )
+    # pylint: disable=unused-argument,invalid-name
+    def user(self, request, user_pk, pk=None):
+        """Get, update, or delete a user of a room."""
+        room = self.get_object()
+        room_user = RoomUser.objects.get(room=room, user__pk=user_pk)
```

Comment:
now I know why it should be called `RoomUserRelation`: it took me 1 minute to understand that this was not a user... The reader a RoomUser is a user in a room...

Reply:
Relation still confuses me… Maybe RoomUserRole ?

Reply:
or `RoomUserAccess` like in [marsha](https://github.com/openfun/marsha)?

Reply:
Yes, good idea.
Let's do this in a dedicated PR.